core/src/main/java/software/amazon/awssdk/core/SdkResponseMetadata.java

```diff
@@ -61,10 +66,14 @@ public ResponseMetadata(ResponseMetadata originalResponseMetadata) {
      *
      * @return The AWS request ID contained in this response metadata object.
      */
-    public String getRequestId() {
+    public String requestId() {
         return metadata.get(AWS_REQUEST_ID);
     }
```

Comment:
I'm not entirely confident about the response metadata abstraction being at the SDK level. Would it be better to code generate it into the model for each of the AWS services? We could add a common `AwsResponseMetadataAware` interface if people want to handle it generically.

Reply:
This is what I'm thinking too since different services may have different metadata. For the JSON case it really just needs to be a facade around the HTTP metadata, i.e. just calls getHeader(x-amz-request-id). For XML we'd need something a bit fancier since the metadata is within the XML document itself so we'd need that map of extracted metadata produced by the unmarshaller.

Reply:
Yeah, I thought about that. The reason I decided to put it at SDK level is that API Gateway API also returns a `x-amzn-RequestId` header and `requestId()` is an interface in `SdkResponseMetadata` and if we remove it out from `SdkResponse`, the generated apig clients' response classes will not have the `requestId()` method anymore and customers would need to pull the requestId header by themselves and to solve that, we might end up creating this SdkResponseMetadata which would do the same thing, extracting `x-amazn-RequestId` from the header.

But we could create another `AwsResponseMetadata` thing in AwsResponse

Reply:
I think they will eventually diverge so we can push the APIG relevant metadata down to it's own base response class. For example APIG also returns some cloudfront-id that we could model if we wanted to. Yes it'll probably be a bit of duplication for the request id but I think ultimately that buys us the most flexibility in evolving the interfaces separately.